Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -0,0 +1,834 @@
+    return 'ok'
+
+
+def get_events_command(client: Client, args: Dict) -> CommandResults:
+    """
+    implements the trend-micro-vision-one-get-events command, mainly used for debugging.
+
+    Args:
+        client (Client): the client object.
+        args (dict): the command arguments.
+
+    Returns:
+        CommandResults: command results object.
+    """
+    limit = arg_to_number(args.get('limit')) or DEFAULT_MAX_LIMIT
+    should_push_events = argToBoolean(args.get('should_push_events', False))
+    log_type = args.get('log_type') or 'all'
+    from_time = args.get('from_time')
+    to_time = args.get('to_time') or datetime.now().strftime(DATE_FORMAT)
+
+    def parse_workbench_logs() -> List[Dict]:
+        workbench_logs = client.get_workbench_logs(
+            start_datetime=from_time,  # type: ignore[arg-type]
+            end_datetime=to_time,
+            limit=limit
+        )
+        return [
+            {
+                'Id': log.get('id'),
+                'Time': log.get('createdDateTime'),
+                'Type': 'Workbench',
+            } for log in workbench_logs
+        ]
+
+    def parse_observed_attack_techniques_logs() -> List[Dict]:
+        observed_attack_techniques_logs = client.get_observed_attack_techniques_logs(
+            detected_start_datetime=from_time,  # type: ignore[arg-type]
+            detected_end_datetime=to_time,  # type: ignore[arg-type]
+            top=limit,
+            limit=limit
+        )
+        return [
+            {
+                'Id': log.get('uuid'),
+                'Time': log.get('detectedDateTime'),
+                'Type': 'Observed Attack Technique'
+            } for log in observed_attack_techniques_logs
+        ]
+
+    def parse_search_detection_logs() -> List[Dict]:
+        search_detection_logs = client.get_search_detection_logs(
+            start_datetime=from_time,  # type: ignore[arg-type]
+            end_datetime=to_time,
+            top=limit,
+            limit=limit
+        )
+        return [
+            {
+                'Id': log.get('uuid'),
+                'Time': timestamp_to_datestring(timestamp=log.get('eventTime'), date_format=DATE_FORMAT, is_utc=True),
+                'Type': 'Search Detection'
+            } for log in search_detection_logs
+        ]
+
+    def parse_audit_logs() -> List[Dict]:
+        audit_logs = client.get_audit_logs(
+            start_datetime=from_time,  # type: ignore[arg-type]
+            end_datetime=to_time,
+            limit=limit
+        )
+        return [
+            {
+                'Id': log.get('loggedUser'),
+                'Time': log.get('loggedDateTime'),
+                'Type': 'Audit'
+            } for log in audit_logs
+        ]
+
+    if log_type == 'all':
+        events = parse_workbench_logs() + parse_observed_attack_techniques_logs() + \
+            parse_search_detection_logs() + parse_audit_logs()
+    else:
+        log_type_to_parse_func = {
```

Comment:
the design didn't mentioned to let the users to ability to choose which kind of logs to retrieve when fetching. 
on the `get_events` it can definitely be implemented.

Reply:
@meital-aizen WDYT?

Reply:
In XSIAM we fetch all log types and don't let the user filter or choose which types they want - this is the standard for all our collectors.